torch/distributions/constraints.py

```diff
@@ -0,0 +1,104 @@
+import torch
+from torch.nn.functional import sigmoid, softmax
+
+
+class Constraint(object):
```

Comment:
In a subsequent PR addressing #56 I plan to add `.bijector` attributes to this class. The `.to_*()` methods are aimed at SVI whereas the `.bijector` methods are aimed at HMC.